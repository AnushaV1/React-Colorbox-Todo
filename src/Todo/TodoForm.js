import React, {useState} from "react";
import { v4 as uuid } from 'uuid';
const TodoForm = ({ addTodo }) => {
    const [task, setTask] = useState("");

    const handleChange = (e) => {
        setTask(e.target.value);
        };

    
    const handleSubmit = (e) => {
        e.preventDefault();
        addTodo({task, id: uuid()});
        setTask("");
    }


    return (
    <form onSubmit={handleSubmit}>
    <label htmlFor='task'>Add Task</label>
    <input type="text"
        name="task"
        placeholder="Enter Todo item"
        id="task"
        onChange ={handleChange}
        value = {task}
    />
    <button>Add To List</button>
    </form>

    )
}

export default TodoForm;