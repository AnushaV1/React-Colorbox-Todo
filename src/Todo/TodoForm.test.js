import React from "react";
import { render, fireEvent } from "@testing-library/react";
import TodoForm from './TodoForm';

it('renders without crashing', function() {
    render (<TodoForm />);
})

it('Matches snapshot', function () {
    const {asFragment} =  render (<TodoForm />);
    expect(asFragment()).toMatchSnapshot();
})

it("test create function on form submit", function() {
    const createMock = jest.fn();
    const { getByText } = render(<TodoForm addTodo={createMock} />);
    const btn = getByText("Add To List");
    fireEvent.click(btn);
    expect(createMock).toHaveBeenCalled();
});