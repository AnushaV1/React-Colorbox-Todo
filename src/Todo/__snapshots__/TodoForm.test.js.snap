// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches snapshot 1`] = `
<DocumentFragment>
  <form>
    <label
      for="task"
    >
      Add Task
    </label>
    <input
      id="task"
      name="task"
      placeholder="Enter Todo item"
      type="text"
      value=""
    />
    <button>
      Add To List
    </button>
  </form>
</DocumentFragment>
`;
